src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Analyzer.cs

```diff
@@ -336,6 +337,12 @@ private static bool AnyArgument(ImmutableArray<IArgumentOperation> arguments, Fu
             // Checks if the parameters of the two passed methods only differ in a ct.
             static bool HasSameParametersPlusCancellationToken(ITypeSymbol cancellationTokenType, IMethodSymbol originalMethod, IMethodSymbol methodToCompare)
             {
+                // Overload is not valid if it has a different return type
+                if (!originalMethod.ReturnType.Equals(methodToCompare.ReturnType))
+                {
+                    return false;
+                }
+
                 // Avoid comparing to itself, or when there are no parameters, or when the last parameter is not a ct
                 if (originalMethod.Equals(methodToCompare) ||
```

Comment:
I agree with your change. However, I feel like the cases that were already being checked inside this `if` are more common than the case you're adding. Can you please consider putting your new condition at the end of the existing `if`?